File: c/hamming.c
#include "hamming.h"
#include <stddef.h>

int compute(const char *a, const char *b) {
  if (a == NULL || b == NULL)
    return -1;
  int diff = 0;
  while (*a != '\0' && *b != '\0') {
    if (*a++ != *b++) {
      diff++;
    }
  }
  if (*a != '\0' || *b != '\0')
    return -1;
  return diff;
}

File: lua/hamming.lua
local Hamming = {}

function Hamming.compute(a, b)
    if #a ~= #b then
        return -1
    end

    local ct = 0

    for i = 1, #a do
        if a:byte(i) ~= b:byte(i) then
            ct = ct + 1
        end
    end

    return ct
end

return Hamming
